Python/import.c

```diff
@@ -1700,6 +1802,9 @@ create_builtin(PyThreadState *tstate, PyObject *name, PyObject *spec)
         return NULL;
     }
 
+#ifdef Py_GIL_DISABLED
+    _PyEval_EnableGILTransient(tstate);
+#endif
```

Comment:
I'd expect this in `import_run_extension()`, right before the `_PyImport_RunModInitFunc()` call.  Then the GIL would be re-enabled right after that call.

Reply:
> That was my first thought, but the module init function can be called in `import_find_extension()` (through `reload_singlephase_extension()`), [here](https://github.com/python/cpython/blob/1b639a04cab0e858d90e2ac459fb34b73700701f/Python/import.c#L1789). This call must be before the first possible call to the module init function.

The call in `reload_singlephase_extension()` is only possible if the module was successfully loaded through `import_run_extension()` first.  By the time we reach `reload_singlephase_extension(), we already know the GIL was left enabled if the module needed it so we wouldn't have to try enabling it again.  Perhaps I've misunderstood how that works?

Reply:
> If there are going to be two different calls to these functions anyway, I could put them in `import_run_extension()` and `reload_singlephase_extension()`, instead of `create_builtin()` and `_imp_create_dynamic_impl()`?

If we need that then, yeah, that would make sense.

Reply:
> By the time we reach `reload_singlephase_extension(), we already know the GIL was left enabled if the module needed it so we wouldn't have to try enabling it again.

Good point, I hadn't thought of that. To make sure I'm understanding correctly: The GIL could have been enabled in a different interpreter, but if this module is loaded in both interpreters, they must be sharing the main GIL?

Assuming that's correct, I should be able to move to one call in `import_run_extension()`.

Reply:
> I should be able to move to one call in `import_run_extension()`

While trying to make that change, I found that the flow inside `import_run_extension()` is complicated enough (especially with the [two](https://github.com/python/cpython/blob/1b639a04cab0e858d90e2ac459fb34b73700701f/Python/import.c#L1881-L1882) [different](https://github.com/python/cpython/blob/1b639a04cab0e858d90e2ac459fb34b73700701f/Python/import.c#L1957-L1960) error paths, both of which are after we'd enable the GIL) that I think it makes sense to keep the enable/disable GIL calls in the callers of `import_run_extension()`. I can still move them to more tightly wrap the `import_run_extension()` call.

Reply:
> I think it makes sense to keep the enable/disable GIL calls in the callers of `import_run_extension()`.

That's fine but means to me that we have a deficiency in `import_run_extension()`.  We can sort that out later.  Please add a note to the two spots though.